### PR TITLE
Add details for inspect command usage

### PR DESCRIPTION
**Pull request by vdemeester**
Carrying #31738

---

Updated documentation on docker inspect sub-command.

/cc @thaJeztah @mstanleyjones 

🦁

Signed-off-by: Eric G. Noriega <egnoriega@users.noreply.github.com>
Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: docs/reference/commandline/inspect.md [hunk 1]
```diff
@@ -30,13 +30,26 @@ Options:
...
 describes all the details of the format.
 
+## Specify target type (--type)
+
+`--type container|image|node|network|service|volume|task|plugin`
+
+It has always been possible for a container to be assigned the same name as an image.
+In this case the result of the inspect will target one of these constructs.  Which type
+is given precedence is not guaranteed. In this case, specify type as an argument
+to the request.  In addition inspect can target any construct type, which may further
```
Comment (allencloud) [inline docs/reference/commandline/inspect.md]: I think that we have a double white spaces here in `request.  In addition`. Could us remove one? 😄

**Comment by allencloud** [comment]
I found that secret was not a inspect type, and should we add that? (currently docker/master does not support `docker inspect secret_name`)

File: docs/reference/commandline/inspect.md (hunk 1 above)
Comment (vdemeester) [inline docs/reference/commandline/inspect.md]: 😂

**Comment by vdemeester** [comment]
oohh intersting 😅 (cc @icecrime :trollface:) @allencloud I'll take a look 👼

File: docs/reference/commandline/inspect.md [hunk 2]
```diff
@@ -30,13 +30,26 @@ Options:
 
 ## Description
 
-By default, `docker inspect` will render all results in a JSON array. If the container and
-image have the same name, this will return container JSON for unspecified type.
+Docker inspect provides detailed information on constructs controlled by Docker.
+
+By default, `docker inspect` will render results in a JSON array.
+
+## Request a custom response format (--format)
```
Comment (thaJeztah) [inline docs/reference/commandline/inspect.md]: can you add a blank line after this heading?

File: docs/reference/commandline/inspect.md [hunk 3]
```diff
@@ -30,13 +30,26 @@ Options:
...
 If a format is specified, the given template will be executed for each result.
 
 Go's [text/template](http://golang.org/pkg/text/template/) package
 describes all the details of the format.
 
+## Specify target type (--type)
+
+`--type container|image|node|network|service|volume|task|plugin`
+
+It has always been possible for a container to be assigned the same name as an image.
```
Comment (thaJeztah) [inline docs/reference/commandline/inspect.md]: Would this work (as there's more possibilities than just "images" and "containers" to have the same name);

    The `docker inspect` command matches any type of object by either ID or name.
    In some cases multiple type of objects (for example, a container and a volume)
    exist with the same name, making the result ambigious.

    To restrict `docker inspect` to a specific type of object, use the `--type`
    option.

    The following example inspects a _volume_ named "myvolume"

    ```bash
    $ docker inspect --type=volume myvolume
    ```

**Review by thaJeztah — CHANGES_REQUESTED** [review]
left a suggestion, but let me know what you think 👍

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by boaz0 — APPROVED** [review]
LGTM 2

**Pull request merged**